cmd/osbuild-worker/jobimpl-koji-finalize.go

```diff
@@ -214,8 +314,55 @@ func extractDynamicArgs(job worker.Job) (*worker.KojiInitJobResult, []worker.OSB
 	return &kojiInitResult, osbuildKojiResults, nil
 }
 
+// Extracts dynamic args of the koji-finalize job. Returns an error if they
+// cannot be unmarshalled.
+func extractDynamicArgs(job worker.Job) (*worker.KojiInitJobResult, []worker.OSBuildJobResult, error) {
+	var kojiInitResult worker.KojiInitJobResult
+	err := job.DynamicArgs(0, &kojiInitResult)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	osbuildResults := make([]worker.OSBuildJobResult, job.NDynamicArgs()-1)
+
+	for i := 1; i < job.NDynamicArgs(); i++ {
+		err = job.DynamicArgs(i, &osbuildResults[i-1])
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
+	return &kojiInitResult, osbuildResults, nil
+}
+
+// Tests the first osbuild job result for specific values and decides
+// if it is a `worker.OSBuildKojiJobResult` or `worker.OSBuildJobResult`.
+// In case of `worker.OSBuildKojiJobResult` returns `true`.
+// In case of 'worker.OSBuildJobResult` returns `false`.
+func isOldKojiComposeReq(job worker.Job) (bool, error) {
```

Comment:
If anyone has a better idea how to handle / determine this, I'm open to suggestions 🙄

Reply:
There's more strict unmarshalling: https://pkg.go.dev/encoding/json#Decoder.DisallowUnknownFields

Reply:
oh, nice! Thanks.

Reply:
Yes, `DisallowUnknownFields` was what I was thinking of.  I should be more clear about that in the future :)

Reply:
Because we have custom `UnmarshalJSON()` methods defined for these structures, using `DisallowUnknownFields` turned out to be impossible. After a discussion, we agreed that the current implementation is acceptable. In addition, it is just temporary, because once we deprecate `OSBuildKoji` jobs, it should be removed.